### PR TITLE
[Metrics] Disable high cardinality operation metrics by default

### LINKED ISSUE #29199: [Dashboard] The agent.py process leaks memory
### What happened + What you expected to happen

I raised this issue on the [discussion forum](https://discuss.ray.io/t/large-memory-usesage-by-dashboard/7800).

I’m running some RLlib + Tune workloads on multiple nodes.

After a day or so I’m getting:

```
(_PackActor pid=233066, ip=10.10.4.2) ray._private.memory_monitor.RayOutOfMemoryError: More than 95% of the memory on node ***** is used (357.77 / 376.36 GB). The top 10 memory consu
mers are:                  
(_PackActor pid=233066, ip=10.10.4.2)   
(_PackActor pid=233066, ip=10.10.4.2) PID       MEM     COMMAND
(_PackActor pid=233066, ip=10.10.4.2) 256207    127.58GiB       /usr/local/bin/python3 -u /usr/local/lib/python3.9/site-packages/ray/dashboard/agent.py --node-ip-ad
(_PackActor pid=233066, ip=10.10.4.2) 256148    3.14GiB /usr/local/lib/python3.9/site-packages/ray/core/src/ray/raylet/raylet --raylet_socket_name=/tmp/ray/
(_PackActor pid=233066, ip=10.10.4.2) 256060    1.71GiB /usr/local/lib/python3.9/site-packages/ray/core/src/ray/gcs/gcs_server --log_dir=/tmp/ray/session_20
(_PackActor pid=233066, ip=10.10.4.2) 256351    1.51GiB python3 -u scripts/train.py --logdir logs/hyperion/exp-set-04/exp-01 --exp-name exp-gnn/exp-set-04/e
(_PackActor pid=233066, ip=10.10.4.2) 260405    1.25GiB ray::RolloutWorker
(_PackActor pid=233066, ip=10.10.4.2) 258394    1.24GiB ray::RolloutWorker
(_PackActor pid=233066, ip=10.10.4.2) 260391    1.24GiB ray::RolloutWorker                                                                                                                                         
(_PackActor pid=233066, ip=10.10.4.2) 258389    1.24GiB ray::RolloutWorker
(_PackActor pid=233066, ip=10.10.4.2) 260392    1.24GiB ray::RolloutWorker
(_PackActor pid=233066, ip=10.10.4.2) 258382    1.24GiB ray::RolloutWorker  
```

That doesn’t look healthy.
I can start the worker with `ray start <...> --include-dashboard false` (according to the [doc](https://docs.ray.io/en/latest/ray-core/configure.html#head-node)), but this might be indicating some deeper issue.

### Versions / Dependencies

Ray: 9f1ea307fe3d5be78006eedb3325d8e06a3b50eb
Python: 3.9.7
Ubuntu: 20.04 (Docker base: `nvidia/cuda:11.6.1-cudnn8-devel-ubuntu20.04`)

### Reproduction script

I'm not sure how to reproduce this with a small script, I'm running quite large experiments at the moment on a Slurm cluster.
Currently the simplest setup that shows this issue is the following:
1. 1 GPU node that runs the PPO trainers (`ray::PPO.train()`)
2. 1 CPU node that only runs environment samplers (`ray::RolloutWorker`) and the Ray head.

The OOM happens on the CPU node.
I think a simple Tune + RLlib setup should be able to reproduce this on a single node, I don't have that much custom stuff in my code. But maybe it's the node to node communication that causes the leak.

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Looks like the agent leaks the memory because it is highly overloaded, and the gRPC requests are queued up without being GC;ed,

This PR does 2 things for the short term mitigation
1. Increase the worker metrics export interval (recover to the original values)
2. Remove high cardinality metrics (operation + grpc) from worker metrics

## Related issue number

Closes https://github.com/ray-project/ray/issues/29199 (I already asked the user to try high worker metrics export interval)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/common/ray_config_def.h [hunk 1]
```diff
@@ -485,7 +489,7 @@ RAY_CONFIG(int64_t, idle_worker_killing_time_threshold_ms, 1000)
 RAY_CONFIG(int64_t, num_workers_soft_limit, -1)
 
 // The interval where metrics are exported in milliseconds.
-RAY_CONFIG(uint64_t, metrics_report_interval_ms, 5000)
+RAY_CONFIG(uint64_t, metrics_report_interval_ms, 30000)
```
Comment (rkooo567) [inline src/ray/common/ray_config_def.h:492]: This is a key to fix memory leak in the short term

**Review by ericl — APPROVED** [review]

File: src/ray/common/ray_config_def.h [hunk 2]
```diff
@@ -22,9 +22,13 @@
 RAY_CONFIG(uint64_t, debug_dump_period_milliseconds, 10000)
 
 /// Whether to enable Ray event stats collection.
-/// TODO(ekl) this seems to segfault Java unit tests when on by default?
 RAY_CONFIG(bool, event_stats, true)
 
+/// Whether to enbale Ray event stats metrics export.
+/// Note that enabling this adds high overhead to
+/// Ray metrics agent.
+RAY_CONFIG(bool, event_stats_metrics, false)
```
Comment (rickyyx) [inline src/ray/common/ray_config_def.h:30]: this is an ambiguous naming though. What does "event stats" here mean? Could we doc more on what metrics it controlled?

Comment (rkooo567) [reply src/ray/common/ray_config_def.h:30]: Actually, if you look at line 25, we already use this term (and this is internal usage only. Most of people don't know how to interpret it anyway).

Comment (rickyyx) [reply src/ray/common/ray_config_def.h:30]: > Most of people don't know how to interpret it anyway)

haha, yeah, i am exactly one of those people. I did a grep on the relevant stuff - i guess you are right that it's used pretty universally. I guess it's just that it wasn't documented clear enough what it does in general.

**Pull request merged**